docs/modules/ROOT/pages/governance.adoc

```diff
@@ -197,3 +198,120 @@ pub trait TimelockABI<TState> {
+    use openzeppelin_utils::cryptography::nonces::NoncesComponent;
+    use openzeppelin_utils::cryptography::snip12::SNIP12Metadata;
+    use starknet::ContractAddress;
+
+    component!(path: VotesComponent, storage: erc20_votes, event: ERC20VotesEvent);
+    component!(path: ERC20Component, storage: erc20, event: ERC20Event);
+    component!(path: NoncesComponent, storage: nonces, event: NoncesEvent);
+
+    // Votes
+    #[abi(embed_v0)]
+    impl VotesImpl = VotesComponent::VotesImpl<ContractState>;
+    impl VotesInternalImpl = VotesComponent::InternalImpl<ContractState>;
+
+    // ERC20
+    #[abi(embed_v0)]
+    impl ERC20MixinImpl = ERC20Component::ERC20MixinImpl<ContractState>;
+    impl ERC20InternalImpl = ERC20Component::InternalImpl<ContractState>;
+
+    // Nonces
+    #[abi(embed_v0)]
+    impl NoncesImpl = NoncesComponent::NoncesImpl<ContractState>;
+
+    #[storage]
+    pub struct Storage {
+        #[substorage(v0)]
+        pub erc20_votes: VotesComponent::Storage,
+        #[substorage(v0)]
+        pub erc20: ERC20Component::Storage,
+        #[substorage(v0)]
+        pub nonces: NoncesComponent::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        ERC20VotesEvent: VotesComponent::Event,
+        #[flat]
+        ERC20Event: ERC20Component::Event,
+        #[flat]
+        NoncesEvent: NoncesComponent::Event
+    }
+
+    // Required for hash computation.
+    pub impl SNIP12MetadataImpl of SNIP12Metadata {
+        fn name() -> felt252 {
+            'DAPP_NAME'
+        }
+        fn version() -> felt252 {
+            'DAPP_VERSION'
+        }
+    }
+
+    // We need to call the VotesComponent::transfer_voting_units function 
+    // after every mint, burn and transfer.
+    // For this, we use the ERC20Component::ERC20HooksTrait.
+    impl ERC20VotesHooksImpl<
+        TContractState,
+        impl Votes: VotesComponent::HasComponent<TContractState>,
+        impl HasComponent: ERC20Component::HasComponent<TContractState>,
+        +NoncesComponent::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of ERC20Component::ERC20HooksTrait<TContractState> {
+        fn after_update(
+            ref self: ERC20Component::ComponentState<TContractState>,
+            from: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) {
+            let mut votes_component = get_dep_component_mut!(ref self, Votes);
+            votes_component.transfer_voting_units(from, recipient, amount);
+        }
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        self.erc20.initializer("MyToken", "MTK");
+    }
+}
+----
+
+The VotesComponent will automatically track voting power as tokens are transferred, minted, or burned.
+
+
+
+
+
+
+
+For a detailed API reference, see the xref:api/governance.adoc#VotesComponent[VotesComponent API documentation].
```

Comment:
Yeah.... I don't think it adds a lot of value. So we can rely on the link at the botton of the page. Added the interface section as well.
```
This is the full interface of the `VotesImpl` implementation:
[source,cairo]
----
#[starknet::interface]
pub trait IVotes<TState> {
    // IVotes
    fn get_votes(self: @TState, account: ContractAddress) -> u256;
    fn get_past_votes(self: @TState, account: ContractAddress, timepoint: u64) -> u256;
    fn get_past_total_supply(self: @TState, timepoint: u64) -> u256;
    fn delegates(self: @TState, account: ContractAddress) -> ContractAddress;
    fn delegate(ref self: TState, delegatee: ContractAddress);
    fn delegate_by_sig(ref self: TState, delegator: ContractAddress, delegatee: ContractAddress, nonce: felt252, expiry: u64, signature: Array<felt252>);

    // INonces
    fn nonces(self: @TState, owner: ContractAddress) -> felt252;
}
----
```

Based on other examples we try to include the full abi that the component generates, I have a few quesitons / topics for disucssion
- Is it accurate to say this is the interface of the implementation?
- I saw in many components we have an `IComponent`interface and another `ComponentABI`interface, which has more functions related to other components. I'm not sure I understand the reason to having both and if we should take into account here in the docs and in the component overall

Reply:
The name of this trait should be `VotesABI` and not `IVotes`, since it contains INonces, and we are presenting ABIs in other interface sections. We should also add VotesABI to the interface file under the votes directory, since now we have the component.

ABIs are traits containing all the external functions that must/can be implemented when a component is in use. If SRC5 is a requirement for a component (like AcessControlComponent), then supports_interface should be part of the ABI even when it is not defined in the component itself. When there are optional external functions that may or may not be embedded when using the component, those should be included in the ABI as well. The idea of ABIs is to act as a single dispatcher to interact with the component without having to define multiple different ones.

Reply:
Ok, thanks for the explanation. Went ahead with this for the ABI:

```
/// Common interface to interact with the `Votes` component.
#[starknet::interface]
pub trait VotesABI<TState> {
    // Votes
    fn get_votes(self: @TState, account: ContractAddress) -> u256;
    fn get_past_votes(self: @TState, account: ContractAddress, timepoint: u64) -> u256;
    fn get_past_total_supply(self: @TState, timepoint: u64) -> u256;
    fn delegates(self: @TState, account: ContractAddress) -> ContractAddress;
    fn delegate(ref self: TState, delegatee: ContractAddress);
    fn delegate_by_sig(
        ref self: TState,
        delegator: ContractAddress,
        delegatee: ContractAddress,
        nonce: felt252,
        expiry: u64,
        signature: Array<felt252>
    );

    // Nonces
    fn nonces(self: @TState, owner: ContractAddress) -> felt252;
}
```

> When there are optional external functions that may or may not be embedded when using the component, those should be included in the ABI as well.
In this case ERC20 and ERC721 functions(and their depenedencies) may be part of the public ABI of someone using the VotesComponent, but I imagine we don't want to include all of this here right?

Reply:
> but I imagine we don't want to include all of this here right?
Right